server/verify.go

```diff
@@ -329,24 +330,36 @@ func VerifyGceTechnology(attestation *pb.Attestation, tech pb.GCEConfidentialTec
 	case pb.GCEConfidentialTechnology_AMD_SEV_ES: // Not verifiable on GCE
 		return nil
```

Comment:
return an error if opts is non-nil for all NONE, SEV, and SEV_ES cases. Add a test case for that expecting to verify SEV-SNP on SEV is an error.

Reply:
IIUC, it will detect the hardware technology enabled on the cvm by itself only.